pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -409,12 +412,37 @@ impl ConnectionManagerState {
                     ctx,
                     node_id,
                 )
-                .await
-                .context("walreceiver connection handling failure")
+                .await;
+
+                match res {
+                    Ok(()) => Ok(()),
+                    Err(e) => {
+                        // FIXME: right now handle_walreceiver_connection has some `?` usage and
+                        // some filtering usage, so this is all a bit confusing.
```

Comment:
This should had been removed in https://github.com/neondatabase/neon/pull/4402/commits/50b94de4f7bb8c15f6620cc496f197c2d26bdf7c

Reply:
[6d2829f](https://github.com/neondatabase/neon/pull/4402/commits/6d2829f0eb61358fb1209877dd387247577a7ffd)